pkg/restapi/vc/operation/persistence.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package operation
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+const (
+	keyPattern       = "%s_%s"
+	profileKeyPrefix = "profile"
+)
+
+// NewCredentialRecorder returns new credential recorder instance
+func NewCredentialRecorder(store storage.Store) *CredentialRecorder {
+	return &CredentialRecorder{store: store}
+}
+
+// CredentialRecorder takes care of features to be persisted for credentials
+type CredentialRecorder struct {
+	store storage.Store
+}
+
+// SaveProfile saves issuer profile to underlying store
+//
+// Args:
+//
+//
+// Returns:
+//
+// error: error
+func (c *CredentialRecorder) SaveProfile(profileResponse *ProfileResponse) error {
+	k := fmt.Sprintf(keyPattern, profileKeyPrefix, profileResponse.ID)
+	bytes, err := json.Marshal(profileResponse)
+
+	if err != nil {
+		return fmt.Errorf("save profile marshalling error: %s", err.Error())
+	}
+
+	return c.store.Put(k, bytes)
+}
+
+// GetProfile returns profile id for given key from underlying store and
+// stores the result in the value pointed to by v
+//
+// Args:
+//
+// id: profile id
+//
+// Returns:
+//
+// profile found
+// error: error
+func (c *CredentialRecorder) GetProfile(id string) (*ProfileResponse, error) {
```

Comment:
Hmm, getting something from a "recorder" type seems a bit funny to me. Perhaps another name should be chosen